_includes/tutorials/multiple-event-types/kafka/markup/dev/make-multi-event-app.adoc

```diff
@@ -0,0 +1,147 @@
+
+The second point is that you can use auto-registration feature of Scheama Registry with Protobuf and the referenced schemas get registered recursively.
+
+Let's take a look at the Avro producer code now:
+[source, java]
+.KafkaProducer for Avro events
+----
+public void produceAvroEvents(final Supplier<Producer<String, SpecificRecordBase>> producerSupplier,
+                              final String topic,
+                              final List<SpecificRecordBase> avroEvents) {
+
+        try (Producer<String, SpecificRecordBase> producer = producerSupplier.get()) {  <1>
+           avroEvents.stream()  <2>
+                    .map((event -> new ProducerRecord<>(topic, (String) event.get("customer_id"), event))) <3>
+                    .forEach(producerRecord -> producer.send(producerRecord,
+//Details left out for clarity
+
+//Relevant configurations
+
+ avroProduceConfigs.put(ProducerConfig.VALUE_SERIALIZER_CLASS_CONFIG, KafkaAvroSerializer.class);
+ avroProduceConfigs.put(AbstractKafkaSchemaSerDeConfig.AUTO_REGISTER_SCHEMAS, false); <4>
+ avroProduceConfigs.put(AbstractKafkaSchemaSerDeConfig.USE_LATEST_VERSION, true) <5>
+----
+
+<1> Getting the producer from the supplier
+<2> Streaming over the provided collection of Avro events to send
+<3> Creating the `ProducerRecord` instance, note the use of map-like access to get the required field for the key
+<4> Specifying **_to disable_** automatic schema registation
+<5> Setting to use the latest schema
+
+You have a very similar approach with the Avro producer, but take a took at the type at annotation one - it's an abstract class, `SpecificRecordBase` that every Avro generated class inherits from.  Since the schema for the Avro multi-event topic uses a `union` at the top level, you don't know the concrete type.  Since you want to use the customer id as the key you need to access the field in a map-like fashion by using the field name as it exists in the schema.  This is possible because `SpecificRecordBase` implements the `GenericRecord` interface which provides the `get` method for retrieving a field value by name.
+
+But the biggest difference is the configurations you provide to the producer for the Avro serializer. namely disabling automatic schema registration, otherwise it would override the union schema as the latest one.  Additionally since you've set `use.latest.version` to `true` the serializer looks up the latest version, the union schema, and will use that for serialization. https://www.confluent.io/blog/multiple-event-types-in-the-same-kafka-topic/#avro-unions-with-schema-references[This blog post] by Robert Yokota explains this mechanism in detail.
```

Comment:
```suggestion
But the biggest difference is the configurations you provide to the producer for the Avro serializer, namely disabling automatic schema registration, otherwise it would override the union schema as the latest one.  Additionally since you've set `use.latest.version` to `true` the serializer looks up the latest version, the union schema, and will use that for serialization. https://www.confluent.io/blog/multiple-event-types-in-the-same-kafka-topic/#avro-unions-with-schema-references[This blog post] by Robert Yokota explains this mechanism in detail.
```